project/generate-dockerfile.sh

```diff
@@ -0,0 +1,264 @@
+		       && make device-mapper \
+		       && make install_device-mapper
+	EOF
+else
+	cat <<-'EOF'
+		RUN cd /usr/local/lvm2 \
+		       && ./configure --enable-static_link \
+		       && make device-mapper \
+		       && make install_device-mapper
+		# see https://git.fedorahosted.org/cgit/lvm2.git/tree/INSTALL
+	EOF
+fi
+
+cat <<-'EOF'
+
+# Install lxc
+ENV LXC_VERSION 1.0.7
+RUN mkdir -p /usr/src/lxc \
+       && curl -sSL https://linuxcontainers.org/downloads/lxc/lxc-${LXC_VERSION}.tar.gz | tar -v -C /usr/src/lxc/ -xz --strip-components=1
+RUN cd /usr/src/lxc \
+       && ./configure \
+       && make \
+       && make install \
+       && ldconfig
+
+EOF
+
+if [ "$GCCGO" == "true" ]; then 
+	cat <<-'EOF'
+	       	ENV     GOPATH  /go:/go/src/github.com/docker/docker/vendor
+	EOF
+else
+	cat <<-'EOF'
+		# Install Go
+		ENV GO_VERSION 1.4.2
+		RUN curl -sSL https://golang.org/dl/go${GO_VERSION}.src.tar.gz | tar -v -C /usr/local -xz \
+		       && mkdir -p /go/bin
+		ENV PATH /go/bin:/usr/local/go/bin:$PATH
+		ENV GOPATH /go:/go/src/github.com/docker/docker/vendor
+		RUN cd /usr/local/go/src && ./make.bash --no-clean 2>&1
+    
+		# Compile Go for cross compilation
+		ENV DOCKER_CROSSPLATFORMS \
+		       linux/386 linux/arm \
+		       darwin/amd64 darwin/386 \
+		       freebsd/amd64 freebsd/386 freebsd/arm \
+		       windows/amd64 windows/386
+
+		# (set an explicit GOARM of 5 for maximum compatibility)
+		ENV GOARM 5
+		RUN cd /usr/local/go/src \
+	        && set -x \
+	        && for platform in $DOCKER_CROSSPLATFORMS; do \
+	                GOOS=${platform%/*} \
+	                GOARCH=${platform##*/} \
+	                       ./make.bash --no-clean 2>&1; \
+	        done
+
+		# We still support compiling with older Go, so need to grab older "gofmt"
+		ENV GOFMT_VERSION 1.3.3
+		RUN curl -sSL https://storage.googleapis.com/golang/go${GOFMT_VERSION}.$(go env GOOS)-$(go env GOARCH).tar.gz | tar -C /go/bin -xz --strip-components=2 go/bin/gofmt
+	EOF
+fi
+
+cat <<-'EOF'
+
+# Grab Go's cover tool for dead-simple code coverage testing
+RUN go get golang.org/x/tools/cmd/cover
+
+# TODO replace FPM with some very minimal debhelper stuff
+EOF
+if [ "$GCCGO" == "true" ]; then
+	cat <<-'EOF'
+		RUN apt-get install -y libffi-dev
+	EOF
+fi
+cat <<-'EOF'
+RUN gem install --no-rdoc --no-ri fpm --version 1.3.2
+
+# Get the "busybox" image source so we can build locally instead of pulling
+EOF
+
+if [ "$ARCH" == "ppc64le" ]; then 
+	cat <<-'EOF'
+		RUN     git clone -b docker-test-image-`uname -m` https://github.com/brahmaroutu/power_docker_image.git /docker-test-image
+		RUN     mkdir -p /var/lib/docker && tar -C /var/lib/docker -xf /docker-test-image/unit-tests.tar
+		RUN     git clone  https://github.com/brahmaroutu/power_docker_busybox.git /docker-busybox
+	EOF
+elif [ "$ARCH" == "s390x" ]; then
+	cat <<-'EOF'
+		RUN     git clone -b docker-test-image-`uname -m` https://github.com/brahmaroutu/power_docker_image.git /docker-test-image
```

Comment:
I realize this is probably a leftover of where you started, but the name of your git project is confusing given it is used for both ppc64le and s390x; same for the "power_docker_busybox" tree.  I don't know how simple that might be to fix, or do we want these to put in a more official location?